Pocket/pocket-ios
# Premium Subscriptions: read subscriptions from the App Store

## Summary
* This PR adds the capability of reading subscription prices from the App Store

## Implementation Details
* Update `Info.plist` and `Keys.swift` with subscription Id keys
* Add `PremiumSubscription`, `PremiumSubscriptionStore` and `PremiumUpgradeViewModel` types
* Update `AccountViewModel` and `SettingsView` to inject the view model (and the store) into `PremiumUpgradeViewModel`
* Update `PremiumUpgradeView` to use the view model 

> **Note**
- `StoreKit` will cache the products for short periods of time. So, if you dismiss the Premium Membership screen and then re-open it, you might see names and prices right away instead of the skeletons. This is normal
- This PR only deals with fetching subscriptions from the App Store. Purchasing and cancellation flows will be handled in future PRs

## Test Steps
* checkout, build and run this branch
* log in to Pocket with a free user
* choose the Settings tab
* tap Go Premium
* make sure that, after a loading time (which will depend on network conditions) subscription names and prices are displayed in the buttons below the premium features list (see animations below for more details)
* make sure that, during the loading time, skeletons are displayed in place of subscription names and prices (see animations below for more details)

## PR Checklist:
- [ ] ~~Added Unit / UI tests~~ `StoreKit` unit tests configuration is still wip, will be added in another PR

## Screenshots

iPhone light | iPad dark
-- | --
<img height=450 src=https://user-images.githubusercontent.com/34376330/218870993-be36e77e-6922-47c9-927b-9c3f85f3d8e7.gif> | <img height=450 src=https://user-images.githubusercontent.com/34376330/218871079-ea7bf50d-e291-4f7e-a61e-472c5f1952c8.gif>
